c7n/resources/glue.py

```diff
@@ -184,6 +184,58 @@ class resource_type(TypeInfo):
     augment = universal_augment
 
 
+@GlueCrawler.filter_registry.register('security-config')
+class GlueCrawlerSecurityConfigFilter(ValueFilter):
+    """Filters glue crawlers with security configurations
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: need-kms-cloudwatch
+                resource: glue-crawler
+                filters:
+                  - type: security-config
+                    key: EncryptionConfiguration.CloudWatchEncryption.CloudWatchEncryptionMode
+                    op: ne
+                    value: SSE-KMS
+    """
+
+    permissions = ('glue:GetSecurityConfiguration',)
+    sec_conf_attribute = 'c7n:SecurityConfiguration'
+    boto_security_config_key = 'CrawlerSecurityConfiguration'
+    schema = type_schema('security-config', rinherit=ValueFilter.schema)
+
+    def process(self, resources, event=None):
+        self.log.debug("fetching security configuration for %d glue crawlers" % len(resources))
+        resources = self.get_security_configuration(resources)
+        return super(ValueFilter, self).process(resources, event)
+
+    def __call__(self, r):
+        if self.sec_conf_attribute not in r:
+            return False
+        return self.match(r[self.sec_conf_attribute])
+
+    def get_security_configuration(self, resources):
+        client = local_session(self.manager.session_factory).client('glue')
+        for r in resources:
+            try:
+                security_config_name = r.get(self.boto_security_config_key)
+                if security_config_name is None:
+                    # I can't make a API to get_security_configuration without
+                    # a Name parameter
+                    continue
+                security_configuration_result = client.get_security_configuration(
```

Comment:
please minimize the try except block around the api call instead the block of unrelated to the exception handling code.